games-engines/openscenegraph/openscenegraph-3.6.0.recipe

```diff
@@ -0,0 +1,196 @@
+DESCRIPTION="The OpenSceneGraph is an open source high performance 3D \
+graphics toolkit, used by application developers in fields such as visual \
+simulation, games, virtual reality, scientific visualization and modelling. \
+Written entirely in Standard C++ and OpenGL it runs on all Windows platforms, \
+OSX, GNU/Linux, IRIX, Solaris, HP-Ux, AIX and FreeBSD operating systems. The \
+OpenSceneGraph is now well established as the world leading scene graph \
+technology, used widely in the vis-sim, space, scientific, oil-gas, games and \
+virtual reality industries."
+HOMEPAGE="http://www.openscenegraph.org/"
+COPYRIGHT="2002-2018 Robert Osfield and others"
+LICENSE="OSGPL"
+REVISION="1"
+SOURCE_URI="https://github.com/openscenegraph/OpenSceneGraph/archive/OpenSceneGraph-$portVersion.tar.gz"
+CHECKSUM_SHA256='6f57134ea74a39f1c7b24c285e6278cf906c47f6c681573b86d12173a466efed'
+SOURCE_DIR="OpenSceneGraph-OpenSceneGraph-$portVersion"
+PATCHES="openscenegraph-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion=$portVersion
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	openscenegraph$secondaryArchSuffix = $portVersion
+	lib:libOpenThreads$secondaryArchSuffix = $libVersionCompat
+	lib:libosg$secondaryArchSuffix = $libVersionCompat
+	lib:libosgAnimation$secondaryArchSuffix = $libVersionCompat
+	lib:libosgDB$secondaryArchSuffix = $libVersionCompat
+	lib:libosgFX$secondaryArchSuffix = $libVersionCompat
+	lib:libosgGA$secondaryArchSuffix = $libVersionCompat
+	lib:libosgManipulator$secondaryArchSuffix = $libVersionCompat
+	lib:libosgParticle$secondaryArchSuffix = $libVersionCompat
+	lib:libosgPresentation$secondaryArchSuffix = $libVersionCompat
+	lib:libosgShadow$secondaryArchSuffix = $libVersionCompat
+	lib:libosgSim$secondaryArchSuffix = $libVersionCompat
+	lib:libosgTerrain$secondaryArchSuffix = $libVersionCompat
+	lib:libosgText$secondaryArchSuffix = $libVersionCompat
+	lib:libosgUI$secondaryArchSuffix = $libVersionCompat
+	lib:libosgUtil$secondaryArchSuffix = $libVersionCompat
+	lib:libosgViewer$secondaryArchSuffix = $libVersionCompat
+	lib:libosgVolume$secondaryArchSuffix = $libVersionCompat
+	lib:libosgWidget$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	mesa$secondaryArchSuffix
+#	lib:libavcodec$secondaryArchSuffix
+	lib:libcairo$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libEGL$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libgdal$secondaryArchSuffix
+	lib:libgl$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgif$secondaryArchSuffix
+#	lib:libgstreamer_1.0$secondaryArchSuffix
+	lib:libhalf$secondaryArchSuffix
+	lib:libIlmImf$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:libjasper$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+#	lib:libOpenCOLLADABaseUtils$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
```

Comment:
How about

    devel:libpng$secondaryArchSuffix >= 16

Reply:
We _could_ do that, but it would just be another way to do the same thing as with `devel:libpng16`.
The reasons why I think `devel:libpng16` is better than `devel:libpng >= 16` are:
1) most of our recipes are already using `{lib,devel}:libpng16`. (OK, that's not a valid reason :)
2) `devel:libpng16` and `lib:libpng16` are slightly more `grep`-friendly :)

That said, I don't object a switch to `devel:libpng >= 16`.

Reply:
Once all of our recipes has been tested with png 16, we should consider deprecating png 15 so that we don't have to keep adding a version suffix to `libpng`

Reply:
That's not how it works. There will be a libpng17, etc. So we should always keep the suffix. And it's not easy to deprecate this because we don't know what else is using the packages (even outside our repos).

The version suffix is not added just for fun or to annoy porters. It's added because the libs are ABI incompatible and we do need both versions side by side, so old apps continue to run when they are built against old versions of the lib.

Ideally, native apps would use the translation kit instead of libpng directly and we wouldn't hit such problems. But this only works for native apps.

Reply:
> By deprecation I mean that we should rename the current libpng to libpng15, and libpng16 to libpng, which would promote the usage of the newer libpng (for people who just type pkgman install libpng, which I believe would be quite common, especially if they came from *nix). 

That would be nice, but can we get updates to work as expected? Wouldn't it be quite a mess of REPLACES and stuff like that to make sure the right thing happens?

I would use only versionned names for the packages (so libpng15 and libpng16 and no "libpng" package at all). At least until we can fix the package manager to allow multiple packages with the same name to be installed side by side (this is a limitation in libsolv, but not in the packagefs). Once this is fixed on Haiku side we can then have both packages named "libpng" and installable side by side if needed.

Reply:
I second @pulkomandy.
BTW, as you can imagine, the reason why I think it is better to continue using `lib:libpng16` and `devel:libpng16` is that this will allow a safe and smooth transition from libpng16 to libpng17, since each recipe can be "switched" to the newer libpng independently of all the other recipes.

Whereas transitions such as { `libpng`→`libpng15` &  `libpng16`→ `libpng` } would require a lot of attention (and work) in a very short period of time to check that every recipe which depended on a given branch of libpng also works fine with the newer branch.

Reply:
> Whereas transitions such as { `libpng`→`libpng15` &  `libpng16`→ `libpng` } would require a lot of attention (and work) in a very short period of time to check that every recipe which depended on a given branch of libpng also works fine with the newer branch.

The [ABI Tracker](https://abi-laboratory.pro/index.php?view=upstream-tracker) would help in this case. But I can imagine the amount work that we would need to do when bumping a library used by lots of programs.

So while I believe the scheme I proposed while gives us some conveniency of lesser work to do when `SONAME` is bumped, and/or allowing builds with whatever version of the lib that the user have, doesn't really scale with our lack of workforce to cover the additional maintenance costs, so I believe it's fine that we continue using the current scheme.